atlasdb-config/src/main/java/com/palantir/atlasdb/config/AtlasDbConfig.java

```diff
@@ -271,6 +272,16 @@ public int getDefaultLockTimeoutSeconds() {
      */
     public abstract Optional<TimestampCache> timestampCache();
 
+    /**
+     * If set, a {@link com.palantir.dialogue.clients.DialogueClients.ReloadingFactory} that a
+     * {@link com.palantir.atlasdb.transaction.api.TransactionManager} based on this configuration should use.
+     * This may be useful for ensuring that connection pools are shared between multitenant services, for instance.
+     *
+     * WARNING: Providing reloading factories to multiple distinct TimeLock servers may result in undefined
+     * behaviour.
+     */
+    public abstract Optional<DialogueClients.ReloadingFactory> reloadingFactory();
```

Comment:
AtlasDbConfig is the wrong place, this should really be in TransactionManagers.

Reply:
AtlasDbConfig is json deserializable.

Reply:
Hmm, yeah it did feel like a better `TransactionManagers` fit, though I originally dumped it in `AtlasDbConfig` because the shared timestamp cache is there. I’ll move this across